orderbook/src/database/orders.rs

```diff
@@ -13,14 +13,14 @@ use std::convert::TryInto;
 
 /// Any default value means that this field is unfiltered.
 #[derive(Default)]
-pub struct OrderFilter<'a> {
+pub struct OrderFilter {
```

Comment:
While working on the PR I realized that forcing this struct to contain references makes it annoying to use (because we construct the filter in warp) so I changed it to owned values.

Reply:
It's a bit unfortunate that this change causes a lot of touched lines that are unrelated to the test refactoring. Maybe next time it can be broken up into a separate PR?

Reply:
Sure it could. I thought this was small enough and it being motivated by the other change that it wasn't worth it to make a separate PR but I'll calibrate my threshold for this down.